pkg/scrape/virtualrealporn.go

```diff
@@ -34,10 +34,14 @@ func VirtualRealPornSite(wg *sync.WaitGroup, updateSite bool, knownScenes []stri
 
 		var tmpCast []string
 
-		// Scene ID - get from URL
-		e.ForEach(`link[rel=shortlink]`, func(id int, e *colly.HTMLElement) {
-			sc.SiteID = strings.Split(e.Attr("href"), "?p=")[1]
-			sc.SceneID = slugify.Slugify(sc.Site) + "-" + sc.SiteID
+		e.ForEach(`body`, func(id int, e *colly.HTMLElement) {
```

Comment:
There is only every one <body> in a valid HTML document. There's no need for a ForEach.

Also, I'm not a huge fan of this method of getting the id as it relies on a class name.

There's a better way:

favScript := e.ChildText(".add-to-favs > script")

That should leave you with favScript being "var stars_post = '121466';"

Just parse the post out of that string. You could also parse the deovr javascript which might even be more reliable as it's less likely to change formats vs something VRP controls.

Reply:
That's actually my first preference as well, however I couldn't think of an easy way to parse the Javascript variable. I know that `colly` does not execute Javascript so we can't get those variables out automatically, so we do have to parse them ourselves. Do you have suggestions? regexp is always an option but it's always the last thing I want to use (cue [xkcd](https://xkcd.com/1171/))

Reply:
How about...

```
		e.ForEach(`script[id="deovr-js-extra"]`, func(id int, e *colly.HTMLElement) {
			var jsonObj map[string]interface{}
			jsonData := e.Text[strings.Index(e.Text, "{") : len(e.Text)-12]
			json.Unmarshal([]byte(jsonData), &jsonObj)

			sc.SiteID = jsonObj["post_id"].(string)

```

Reply:
Thanks for the guidance, I'll push some changes soon.